core/state_transition.go

```diff
@@ -134,6 +135,23 @@ func ApplyMessage(evm *vm.EVM, msg Message, gp *GasPool) ([]byte, uint64, bool,
 	return NewStateTransition(evm, msg, gp).TransitionDb()
 }
 
+// ApplyIncomingReceipt will add amount into ToAddress in the receipt
+func ApplyIncomingReceipt(db *state.DB, cxp *types.CXReceiptsProof) {
+	if cxp == nil {
+		return
+	}
+
```

Comment:
I thought here some verification is needed before we add the balance?

Reply:
The verification is done before put into the receiptsList. I think it should be good to go